src/java.base/share/classes/javax/crypto/KDF.java

```diff
@@ -0,0 +1,639 @@
+        throws NoSuchAlgorithmException, NoSuchProviderException {
+        Objects.requireNonNull(algorithm, "algorithm should not be null");
+        Objects.requireNonNull(provider, "provider should not be null");
+        try {
+            return getInstance(algorithm, null, provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "No implementation found using null KDFParameters", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} instance initialized with the specified algorithm from
+     * the specified security provider.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use.
+     *     See the {@code KDF} section in the <a href=
+     *     "{@docRoot}/../specs/security/standard-names.html#kdf-algorithms">
+     *     Java Security Standard Algorithm Names Specification</a>
+     *     for information about standard KDF algorithm names.
+     * @param provider
+     *     the provider to use for this key derivation
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if the specified provider does not support the specified KDF
+     *     algorithm
+     * @throws NullPointerException
+     *     if the {@code algorithm} or {@code provider} is {@code null}
+     */
+    public static KDF getInstance(String algorithm, Provider provider)
+        throws NoSuchAlgorithmException {
+        Objects.requireNonNull(algorithm, "algorithm should not be null");
+        Objects.requireNonNull(provider, "provider should not be null");
+        try {
+            return getInstance(algorithm, null, provider);
+        } catch (InvalidAlgorithmParameterException e) {
+            throw new NoSuchAlgorithmException(
+                "No implementation found using null KDFParameters", e);
+        }
+    }
+
+    /**
+     * Returns a {@code KDF} instance initialized with the specified algorithm and
+     * is initialized with the specified parameters.
+     *
+     * @param algorithm
+     *     the key derivation algorithm to use.
+     *     See the {@code KDF} section in the <a href=
+     *     "{@docRoot}/../specs/security/standard-names.html#kdf-algorithms">
+     *     Java Security Standard Algorithm Names Specification</a>
+     *     for information about standard KDF algorithm names.
+     * @param kdfParameters
+     *     the {@code KDFParameters} used to configure the derivation
+     *     algorithm or {@code null} if no parameters are provided
+     *
+     * @return a {@code KDF} object
+     *
+     * @throws NoSuchAlgorithmException
+     *     if no {@code Provider} supports a {@code KDFSpi} implementation for
+     *     the specified algorithm
+     * @throws InvalidAlgorithmParameterException
+     *     if the initialization parameters are inappropriate for this
+     *     {@code KDF} or if no provider can be found which supports the
```

Comment:
What does "this KDF" mean? This is a static method and there is no instance yet.